activity/activity_PostDigestJATS.py

```
@@ -0,0 +1,130 @@
+
+        # Generate jats
+        self.jats_content = self.digest_jats(self.input_file)
+
+        if self.jats_content:
+            self.statuses["jats"] = True
+
+        # POST to API endpoint
+        try:
+            self.statuses["post"] = self.post_jats(self.digest, self.jats_content)
+        except Exception as exception:
+            self.logger.exception("Exception raised in do_activity. Details: %s" % str(exception))
+
+        self.logger.info(
+            "%s for real_filename %s statuses: %s" % (self.name, str(real_filename), self.statuses))
+
+        if self.statuses.get("jats"):
+            return self.ACTIVITY_SUCCESS
+
+        return self.ACTIVITY_PERMANENT_FAILURE
+
+    def digest_jats(self, docx_file):
+        "generate the digest JATS content from the docx file"
+        try:
+            return digest_provider.build_jats(docx_file, self.temp_dir, self.digest_config)
+        except Exception as exception:
+            self.logger.exception(
+                "Exception generating digest jats for docx_file %s. Details: %s" %
+                (str(docx_file), str(exception)))
+
+    def post_jats(self, digest, jats_content):
+        "prepare and POST jats to API endpoint"
+        url = self.settings.typesetter_digest_endpoint
+        payload = post_payload(digest, jats_content, self.settings.typesetter_digest_api_key)
+        return post_jats_to_endpoint(url, payload)
+
+    def create_activity_directories(self):
+        """
+        Create the directories in the activity tmp_dir
+        """
+        for dir_name in [self.temp_dir, self.input_dir]:
+            try:
+                os.mkdir(dir_name)
+            except OSError:
+                pass
+
+
+def post_payload(digest, jats_content, api_key):
+    "compile the POST data payload"
+    account_key = 1
+    content_type = "digest"
+    payload = OrderedDict()
+    payload["apiKey"] = api_key
+    payload["accountKey"] = account_key
+    if digest:
```

arguably if there is no `digest` this should fail hard, not attempting to do the request at all
(the failure would still be caught in the activity for now)

---

Normally it should be shielded from having no digest because the input is validated in a normal workflow before it reaches this point, but I do have some test cases with bad `.docx` input and this was a quick fix. I agree if there is no `digest` object the payload will be useless, and we can return `None` first if there is no `digest` here 👍